### PR TITLE
Jose catalogue table finished

### PR DESCRIPTION
**Pull request by JoseLozadaAD**

### CODE REVIEW

File: src/controllers/catalogue.controller.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import {
+  createCatalogueUseCase,
+  deleteCatalogueByIdUseCase,
+  getAllCataloguesQuery,
+  getCatalogueByIdQuery,
+  updateCatalogueUseCase
+} from '../repositories/catalogue.repository.js'
+
+export const getAllCatalogues = async (request, response) => {
```
Comment (jacovzap-assuresoft) [inline src/controllers/catalogue.controller.js]: Please follow the convention names of the application using the req, res variables

File: src/routes/catalogue.route.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  createCatalogue,
+  deleteCatalogueById,
+  getAllCatalogues,
+  getCatalogueById,
+  updateCatalogue
+} from '../controllers/catalogue.controller.js'
+
+const catalogueRouter = Router()
+
+catalogueRouter.get('/', getAllCatalogues)
```
Comment (jacovzap-assuresoft) [inline src/routes/catalogue.route.js]: It is not consistent with the way the team manages routing.

**Review by jacovzap-assuresoft — CHANGES_REQUESTED** [review]
Please correct the observations

File: src/repositories/catalogue.repository.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export const createCatalogueUseCase = async data => {
+  return prisma.catalogue.create({
+    data,
+    select: catalogueBody
+  })
+}
+
+export const updateCatalogueUseCase = async ({ data, id }) => {
```
Comment (jacovzap-assuresoft) [inline src/repositories/catalogue.repository.js]: Why send the function arguments inside an object if you're going to destructure them here?

File: src/routes/catalogue.route.js (hunk 2 above)
Comment (Juan-Pablo-Chavez-AS) [reply src/routes/catalogue.route.js]: Please check routing for store.route.js in the main branch @JoseLozadaAD

File: src/routes/catalogue.route.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+catalogueRouter.get('/:id', getCatalogueById)
+catalogueRouter.post('/', createCatalogue)
+catalogueRouter.put('/:id', updateCatalogue)
```
Comment (jacovzap-assuresoft) [inline src/routes/catalogue.route.js]: Put request is not called "updateCatalogueById" as the others requests that recieve the id as param. Also the naming of classes in this file is different from the convention used in the rest of the app

File: src/repositories/catalogue.repository.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
+import prisma from '../config/db.config.js'
+
+const catalogueBody = {
+  catalogue_id: true,
+  type: true,
+  code: true,
+  description: true
+}
+
+export const getAllCataloguesQuery = async () => {
```
Comment (jacovzap-assuresoft) [inline src/repositories/catalogue.repository.js]: Catalogues is not the correct word here, I guess you wanted to say "Catalogs"

**Review by jacovzap-assuresoft — CHANGES_REQUESTED** [review]

File: src/controllers/catalogue.controller.js (hunk 1 above)
Comment (JoseLozadaAD) [reply src/controllers/catalogue.controller.js]: the req and res variables are code smells are not descriptive variables

Comment (jacovzap-assuresoft) [reply src/controllers/catalogue.controller.js]: I let you the documentation of https://expressjs.com/es/guide/routing.html.

They are not code smells, just abbreviation, so you are telling me that using variable "args" in java is code smell?

Comment (josueveliz-assure) [reply src/controllers/catalogue.controller.js:9]: el papeador de papus

File: src/repositories/catalogue.repository.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+export const createCatalogueRepository = async data => {
+  return prisma.catalogue.create({
+    data,
+    select: catalogueBody
+  })
+}
+
+export const updateCatalogueRepository = async ({ data, id }) => {
```
Comment (Juan-Pablo-Chavez-AS) [inline src/repositories/catalogue.repository.js]: Why do you decompose the object that you compose inLine when calling this function?
```javascript
const catalogueUpdated = await updateCatalogueRepository({ data, id: Number(id) })
```

**Comment by Juan-Pablo-Chavez-AS** [comment]
image

**Pull request merged**